lib/pbench/server/api/resources/query_apis/query_dataset.py

```diff
@@ -0,0 +1,268 @@
+from flask import request, jsonify
+from flask_restful import Resource, abort
+import requests
+
+from dateutil import parser
+from pbench.server.api.resources.query_apis import (
+    get_es_url,
+    get_index_prefix,
+    gen_month_range,
+    get_user_term,
+)
+
+
+class QueryDataset(Resource):
+    """
+    Abstracted Pbench API to get detailed data for a particular
+    dataset by name. The date range given isn't used for the search
+    directly, but rather allows building a set of Elasticsearch index
+    names that will contain the data. (Note that the dashboard uses the
+    same date range it used to find controllers; however it could make this
+    more efficient by using only the year and month in which the result
+    was generated.)
+
+    Note that although the name "queryResult" is retained for consistency
+    with the existing dashboard UI code, the word "result" is a bit
+    misleading in that this deals with Pbench "run" documents, not Pbench
+    "results" documents. It's a follow-on to queryResults, which returns
+    the list of datasets within a specified date range for a given controller
+    (as reported by queryControllers). The queryResult API returns details from
+    the dataset run document; basically the union of the "run" and "@metadata"
+    sub-documents.
+    """
+
+    def __init__(self, config, logger):
+        self.logger = logger
+        self.es_url = get_es_url(config)
+        self.prefix = get_index_prefix(config)
+
+    def post(self):
+        """
+        POST to get details for a specific Pbench dataset within the
+        specified date range, and owned by a specified user:
+
+        {
+            "user": "email-tag",
```

Comment:
Why are we requiring that the API invocation provide a "user" argument?

Why not make this parameter optional to add a constraint on which data is visible.  But we should not use that parameter for access control.

Instead, we have to get the effective user of the query, and make sure the query results in data that is "public" or from a user that matches this user.

Reply:
I'm still using the `json_data["user"]` reference here, but I really should change it to `json_data.get("user")` so that a missing `"user"` will simply return as `None` and trigger the query for published data just like an explicit `"user": None`.

And, yes, I could also default it from the authorization token payload, but this still hasn't been hooked up to the authorization decorator. My thought had been to do that in a separate pass.

There's still the basic question of whether we want a logged-in user to be able to see only _their own_ owned data, rather than also all public data. The `"user"` parameter allows disambiguation, and relying on the authorization token I think implies we need a specific parameter if we want to do that. E.g., if I don't specify `"user"` and I'm authorized, do I see just my data, or also all public data? What about an authorized user specifying `"user": None`? Do we default, or take the specification of "anti-user" (public).